### PR TITLE
feat: question delivery — pending UI, chat injection, answer reactions

### PR DESCRIPTION
**Pull request by yanekyuk**
## New Features

- **Pending Questions UI**: Frontend questions service with 30s polling, PendingQuestions panel with sidebar badge and dropdown grouped by source type and mode
- **Question Expiry**: `expiresAt` column with 7-day default TTL on questions; expired questions filtered from pending queries
- **Chat/MCP Question Injection**: `PendingQuestionSummary` type and `findPendingQuestions` on ToolDeps; pending questions merged into `discover_opportunities` tool results with session-level dedup via local Set in ChatAgent
- **Answer Reaction Handlers**: Dispatcher on `QuestionEvents.onAnswered` routes by mode — profile creates a premise (triggers profile regen), intent appends refinement and enqueues HyDE regen, negotiation stores answer in opportunity metadata

## Refactors

- Added `metadata` JSONB column to opportunities table (migration 0076) for answer context storage

## Tests

- 21 handler unit tests (dispatcher, profile, intent, negotiation, wiring)
- Protocol-level tests for pending question merge, dedup, state annotations
- Frontend compile-time type alignment tests
- Edge case coverage for free-text-only answers (empty selectedOptions)

## Test plan

- [ ] 21 handler tests pass (`bun test src/events/handlers/tests/`)
- [ ] Protocol tests pass (`cd packages/protocol && bun test`)
- [ ] Frontend builds (`cd frontend && bun run build`)
- [ ] Verify pending questions appear in sidebar after question generation
- [ ] Verify answering a profile-mode question creates a premise
- [ ] Verify answering an intent-mode question triggers HyDE regeneration
- [ ] Verify expired questions are filtered from the pending list

### CODE REVIEW

File: packages/protocol/src/shared/schemas/pending-question.schema.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  title: string;
+  prompt: string;
+  options: Array<{ label: string; description: string }>;
+  multiSelect: boolean;
+  mode: QuestionMode;
+  sourceType: string;
+  sourceId: string;
+  createdAt: string;
+  expiresAt?: string;
+}
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply packages/protocol/src/shared/schemas/pending-question.schema.ts:19]: PendingQuestionSummary is a read-only projection constructed by our own backend adapter code (questioner.adapter.ts maps DB rows → this shape, mcp.controller.ts maps adapter rows → summary). It never crosses a trust boundary where runtime validation would catch real errors — the producer and consumer are both our typed code. Adding a Zod schema here is YAGNI; we'd add it if this type started being parsed from external input.

File: packages/protocol/src/chat/chat.state.ts [hunk 2]
```diff
@@ -152,6 +152,16 @@ export const ChatGraphState = Annotation.Root({
     default: () => undefined,
   }),
 
+  /**
+   * IDs of questions already surfaced as `decision_questions` events in this
+   * session. Used to deduplicate pending questions injected from tool results
+   * so the same question isn't shown twice within a single chat session.
+   */
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply packages/protocol/src/chat/chat.state.ts]: Fixed. Removed the surfacedQuestionIds state annotation and its test file. Dedup happens via a local Set scoped to each streamRun call — graph state was never read or written.

File: backend/src/events/handlers/question.answer.profile.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+ * Build the assertion text from the answer components.
+ * Joins selected options with "; " and appends freeText if present.
+ */
+function buildAssertionText(selectedOptions: string[], freeText?: string): string {
+  const base = selectedOptions.join('; ');
+  if (freeText?.trim()) {
+    return `${base}. ${freeText.trim()}`;
+  }
+  return base;
+}
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply backend/src/events/handlers/question.answer.profile.ts:34]: Fixed. buildAssertionText now returns just the freeText when selectedOptions is empty, instead of producing a leading ". ". Added a test for the free-text-only path.

File: backend/src/events/handlers/question.answer.intent.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+/**
+ * Build a refinement addendum from the answer.
+ * Format: "[Refined: <options>. <freeText>]"
+ */
+function buildRefinementAddendum(selectedOptions: string[], freeText?: string): string {
+  const parts = selectedOptions.join('; ');
+  const addendum = freeText?.trim() ? `${parts}. ${freeText.trim()}` : parts;
+  return `\n\n[Refined: ${addendum}]`;
+}
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply backend/src/events/handlers/question.answer.intent.ts:33]: Fixed. Same fix applied to buildRefinementAddendum — produces `[Refined: <freeText>]` when selectedOptions is empty. Added a test verifying no leading ". " appears.

File: frontend/src/components/Sidebar.tsx [hunk 5]
```diff
@@ -266,6 +283,30 @@ export default function Sidebar() {
...
+        {pendingQuestionsCount > 0 && (
+          <div className="relative" ref={questionsRef}>
+            <button
+              onClick={() => setQuestionsOpen(!questionsOpen)}
+              className={`w-full flex items-center gap-3 px-3 py-2.5 rounded-md text-sm transition-colors ${
+                questionsOpen
+                  ? 'bg-gray-100 text-black font-bold'
+                  : 'text-black font-medium hover:bg-gray-50'
+              }`}
+            >
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply frontend/src/components/Sidebar.tsx:297]: Fixed. Added `type="button"` and `aria-expanded={questionsOpen}` to the Questions dropdown trigger.

File: backend/src/events/handlers/tests/question.answer.profile.test.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+      userId: "u-1",
+      questionId: "q-1",
+      selectedOptions: ["Solo option"],
+      sourceId: "prof-1",
+    });
+
+    const call = (deps.createPremise as ReturnType<typeof mock>).mock.calls[0][0];
+    expect(call.assertion.text).toBe("Solo option");
+  });
+});
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply backend/src/events/handlers/tests/question.answer.profile.test.ts:115]: Fixed. Added a test case for `selectedOptions: [], freeText: "I prefer async communication"` — verifies the assertion text is just the freeText with no leading separator.

File: backend/src/events/handlers/tests/question.answer.intent.test.ts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    await fn({
+      userId: "u-1",
+      intentId: "int-1",
+      questionId: "q-1",
+      selectedOptions: ["A"],
+    });
+
+    expect(deps.updateIntentDescription).not.toHaveBeenCalled();
+  });
+});
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply backend/src/events/handlers/tests/question.answer.intent.test.ts:137]: Fixed. Added a test case for `selectedOptions: [], freeText` — asserts the addendum contains only the freeText and no leading ". ".

File: backend/src/main.ts [hunk 8]
```diff
@@ -140,6 +146,59 @@ PremiseEvents.onExpired = (premiseId: string, userId: string) => {
     .catch(err => log.job.from('PremiseEvents').error('Failed to enqueue profile regen', { premiseId, userId, error: err }));
 };
 
+// ─── Question answer reaction handlers ──────────────────────────────────────
+
+const questionAnswerDeps = {
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply backend/src/main.ts:151]: Fixed. Updated PR body to use the changelog-section format (New Features, Refactors, Tests, Test plan).

File: frontend/src/contexts/QuestionsContext.tsx [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+      console.error('Failed to fetch pending questions:', err);
+    }
+  }, [questionsService, user?.id]);
+
+  // Initial fetch + polling
+  useEffect(() => {
+    if (!user?.id) {
+      setQuestions([]);
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply frontend/src/contexts/QuestionsContext.tsx:47]: Fixed. Added `setLoading(false)` in the early-return branch when `!user?.id`. Now logout resets both `questions` and `loading`.

File: backend/src/events/handlers/tests/question.answer.wiring.test.ts [hunk 10]
```diff
@@ -0,0 +1,8 @@
+import { describe, it, expect } from "bun:test";
+import { QuestionEvents } from "../../question.event";
+
+describe("QuestionEvents.onAnswered wiring", () => {
+  it("onAnswered is a function (not a no-op after main.ts runs)", () => {
+    expect(typeof QuestionEvents.onAnswered).toBe("function");
+  });
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply backend/src/events/handlers/tests/question.answer.wiring.test.ts]: Removed. The test only checked `typeof` against a no-op initializer — it never imported `main.ts` and would pass even without wiring. Deleted the file.

File: backend/src/adapters/database.adapter.ts [hunk 11]
```diff
@@ -4841,6 +4848,10 @@ export class OpportunityDatabaseAdapter {
     });
   }
 
+  async updateOpportunityMetadata(id: string, metadata: Record<string, unknown>): Promise<void> {
+    await db.update(opportunities).set({ metadata }).where(eq(opportunities.id, id));
+  }
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply backend/src/adapters/database.adapter.ts:4853]: Fixed. Added `updatedAt: new Date()` to the `.set()` call, consistent with `updateOpportunityStatus` and `updateOpportunityActorApproval`.

File: packages/protocol/src/opportunity/opportunity.pending-questions.ts [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+  const filters = input.sourceType
+    ? { sourceType: input.sourceType, ...(input.sourceId ? { sourceId: input.sourceId } : {}) }
+    : undefined;
+
+  let pending: PendingQuestionSummary[];
+  try {
+    pending = await input.findPendingQuestions(input.userId, filters);
+  } catch {
+    return { questions: [], surfacedIds: [] };
+  }
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply packages/protocol/src/opportunity/opportunity.pending-questions.ts:50]: Fixed. Added `console.warn` with the error in the catch block. The protocol package is framework-agnostic (no access to the app logger), so `console.warn` is the appropriate level here.

File: backend/src/controllers/mcp.controller.ts [hunk 13]
```diff
@@ -409,6 +409,21 @@ function getOrCreateMcpServer(): McpServer {
     frontendUrl: protocolDeps.frontendUrl,
     apiBaseUrl: protocolDeps.apiBaseUrl,
     ...(protocolDeps.questionerEnqueue && { questionerEnqueue: protocolDeps.questionerEnqueue }),
+    findPendingQuestions: async (userId: string, filters?: { sourceType?: string; sourceId?: string }) => {
+      const rows = await questionerAdapter.findPending(userId, filters);
+      return rows.map((row): PendingQuestionSummary => ({
+        id: row.id,
+        title: row.payload.title,
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply backend/src/controllers/mcp.controller.ts:416]: `mcp.controller.ts` is the **composition root** — it assembles `ProtocolDeps` and injects them into the protocol layer. This `findPendingQuestions` callback is part of that DI wiring (line 412 builds a closure injected via `ProtocolDeps`), not a controller request handler. The adapter access is intentional here — it's the same pattern used for every other protocol dependency in this file (e.g. `getOpportunity`, `getIntent`, `createPremise`).

File: frontend/src/services/questions.ts [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+) => ({
+  /**
+   * Fetch pending questions for the authenticated user.
+   * Optionally filter by mode, sourceType, or sourceId.
+   */
+  getPending: async (filters?: {
+    mode?: string;
+    sourceType?: string;
+    sourceId?: string;
+  }): Promise<PendingQuestion[]> => {
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply frontend/src/services/questions.ts:69]: Fixed. Narrowed `mode` from `string` to `QuestionDetection['mode']`.

File: backend/drizzle/meta/0075_snapshot.json [hunk 15]
```diff
@@ -0,0 +1,4401 @@
...
+      "uniqueConstraints": {},
+      "policies": {},
+      "checkConstraints": {},
+      "isRLSEnabled": false
+    },
+    "public.opportunities": {
+      "name": "opportunities",
+      "schema": "",
+      "columns": {
+        "id": {
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply backend/drizzle/meta/0075_snapshot.json:2282]: Fixed. Deleted the hand-written SQL, re-ran `db:generate` which produced both the SQL file and `0076_snapshot.json`. Renamed per convention. `db:generate` now reports no changes.

File: packages/protocol/src/opportunity/opportunity.tools.ts [hunk 16]
```diff
@@ -973,7 +987,10 @@ export function createOpportunityTools(defineTool: DefineTool, deps: ToolDeps) {
           ...(result.pagination ? { pagination: result.pagination } : {}),
           debugSteps: allDebugSteps,
           _graphTimings: _allGraphTimings,
-          ...(result.questions && result.questions.length > 0 ? { questions: result.questions } : {}),
+          ...(() => {
+            const allQ = [...(result.questions ?? []), ...pendingQuestions];
+            return allQ.length > 0 ? { questions: allQ } : {};
+          })(),
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply packages/protocol/src/opportunity/opportunity.tools.ts:993]: The caps are per-source by design. `mergePendingQuestions` caps at `MAX_PENDING_QUESTIONS=3` for pending questions, and the discovery graph has its own generation limits for inline questions. These serve different purposes — inline questions are contextual to the current discovery run, pending questions are from earlier contexts. The LLM agent handles final selection and presentation to the user. Capping the merged array would silently discard valid questions from one source when the other is active.

File: packages/protocol/src/opportunity/opportunity.tools.ts [hunk 17]
```diff
@@ -987,7 +1004,10 @@ export function createOpportunityTools(defineTool: DefineTool, deps: ToolDeps) {
...
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply packages/protocol/src/opportunity/opportunity.tools.ts:1010]: Same reasoning as above — per-source caps are intentional.

File: packages/protocol/src/opportunity/opportunity.tools.ts [hunk 18]
```diff
@@ -1007,7 +1027,10 @@ export function createOpportunityTools(defineTool: DefineTool, deps: ToolDeps) {
           summary: "No new matches (existing connections only)",
           debugSteps: allDebugSteps,
           _graphTimings: _allGraphTimings,
-          ...(result.questions && result.questions.length > 0 ? { questions: result.questions } : {}),
+          ...(() => {
+            const allQ = [...(result.questions ?? []), ...pendingQuestions];
+            return allQ.length > 0 ? { questions: allQ } : {};
+          })(),
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply packages/protocol/src/opportunity/opportunity.tools.ts:1033]: Same reasoning as above — per-source caps are intentional.

File: packages/protocol/src/opportunity/opportunity.tools.ts [hunk 19]
```diff
@@ -1188,7 +1211,10 @@ export function createOpportunityTools(defineTool: DefineTool, deps: ToolDeps) {
               suggestedIntentDescription: searchQuery,
             }
           : {}),
-        ...(result.questions && result.questions.length > 0 ? { questions: result.questions } : {}),
+        ...(() => {
+          const allQ = [...(result.questions ?? []), ...pendingQuestions];
+          return allQ.length > 0 ? { questions: allQ } : {};
+        })(),
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply packages/protocol/src/opportunity/opportunity.tools.ts:1217]: Same reasoning as above — per-source caps are intentional.

File: backend/src/events/handlers/question.answer.profile.ts [hunk 20]
```diff
@@ -0,0 +1,86 @@
...
+    freeText?: string;
+    sourceId: string;
+  }): Promise<void> => {
+    const assertionText = buildAssertionText(input.selectedOptions, input.freeText);
+
+    logger.verbose('Creating premise from profile answer', {
+      userId: input.userId,
+      questionId: input.questionId,
+      assertionLength: assertionText.length,
+    });
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply backend/src/events/handlers/question.answer.profile.ts:58]: Fixed. Added early return when `assertionText` is empty. Test added for the edge case (empty selectedOptions + whitespace-only freeText).

File: backend/src/events/handlers/question.answer.intent.ts [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+export function enqueueIntentRefinementFactory(deps: IntentRefinementDeps) {
+  return async (input: {
+    userId: string;
+    intentId: string;
+    questionId: string;
+    selectedOptions: string[];
+    freeText?: string;
+  }): Promise<void> => {
+    const intent = await deps.getIntent(input.intentId);
+
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply backend/src/events/handlers/question.answer.intent.ts:44]: Fixed. Added early return when both `selectedOptions` is empty and `freeText` is whitespace-only, before building the addendum. Test added.
